src/Fixer/Operator/TernaryToElvisOperatorFixer.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Operator;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Gabriel Caruso <carusogabriel34@gmail.com>
+ */
+final class TernaryToElvisOperatorFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Converts a ternary operation with first and second operands identicals to a elvis operatorion (?:).',
+            [
+                new CodeSample(
+                    "<?php \$foo = \$bar ? \$bar : baz;\n"
+                ),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound(['?', ':']);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->equals('?')) {
+                continue;
+            }
+
+            $prevTokenIndex = $tokens->getPrevMeaningfulToken($index);
+            $prevToken = $tokens[$prevTokenIndex];
+
+            $nextTokenIndex = $tokens->getNextMeaningfulToken($index);
+            $nextToken = $tokens[$nextTokenIndex];
+
+            if (!$prevToken->equals($nextToken)) {
+                continue;
+            }
+
+            // previous and before content of ternary operation are identical, continue...
+            $tokens->clearRange($nextTokenIndex - 1, $nextTokenIndex + 1);
```

Comment:
in continuation of
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3523#discussion_r167305641

Let us describe what is what:
`$foo ? $foo : $bar`
let us assume terms for it:
`condition ? val_if_yes : val_if_no`.

So, here, you drop everything around `val_if_yes`, which could be... whitespace? comment? `?` symbol, like in case `Minimal number of tokens case.`. That's why tests are currently failing.

IMHO, this new rule shall only drop `val_if_yes`, doing
```diff
-$foo ? $foo : $bar
+$foo ? : $bar
```

Yes, it could be ugly. But it also won't mess the comments, won't drop real code, and won't mess up with whitespaces.
But hey, you know what? for all that handling of dirty whitespaces, we do have `TernaryOperatorSpacesFixer` already !
And it does know how to deal with multiline whitespaces, so you don't need to take care about it.
And in the end, we write fixers to be idempotent (see https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.10/doc/COOKBOOK-FIXERS.md#background )

So what to do?
1. update src to clear only what is needed
```diff
-$tokens->clearRange($nextTokenIndex - 1, $nextTokenIndex + 1);
+$tokens->clearTokenAndMergeSurroundingWhitespace($nextTokenIndex);
```
2. add multiline test case (don't worry, it doesn't need to look perfectly, it is only to prove that given fixer is not messing up the code, like with current `Minimal number of tokens case.`. to look pretty in this "complex" case, see step 3)
3. add priority integration test that new fixers run before `TernaryOperatorSpacesFixer` and how would look output of code fixed by our 2 fixers one after another


I hope this will help you a bit !

Reply:
@keradus Thanks very much for this explanation, it healed a lot of doubts that I had. Gonna work on that and your suggestions :construction: